monai/visualize/class_activation_maps.py

```diff
@@ -0,0 +1,160 @@
+
+    def norm_features(self, feature):
+        feature -= feature.min()
+        feature /= feature.max()
+        return 1.0 - feature
+
+    def __call__(self, x, class_idx=None):
+        logits, acti, _ = self.net(x)
+        acti = acti[0]
+        if class_idx is None:
+            class_idx = torch.argmax(logits, dim=1)
+        b, c, *spatial = acti.shape
+        weight = self.net.model.fc.weight[class_idx].view(c, *[1 for _ in spatial])
+        map = [self.norm_features((weight * a).sum(0)) for a in acti]
+        return torch.stack(map, dim=0)
+
+
+class GradCAM:
+    def __init__(self, model, target_layers):
+        self.net = ModelWithHooks(model, target_layers, register_forward=True, register_backward=True)
+
+    def norm_features(self, feature):
+        feature -= feature.min()
+        feature /= feature.max()
+        return 1.0 - feature
+
+    def __call__(self, x):
+        logits, acti, grad = self.net(x)
+        acti = acti[0]
+        grad = grad[0]
+        b, c, *spatial = grad.shape
+        grad_ave = grad.view(b, c, -1).mean(2)
+        weights = grad_ave.view(b, c, 1, 1)
+        map = (weights * acti).sum(1)
+        return self.norm_features(map)
+
+
+# if __name__ == "__main__":
+#     from torchvision import transforms
+#     import glob
+#     import numpy as np
+#     import PIL
+#     import cv2
+#     from matplotlib import pyplot as plt
+#     import argparse
+#
+#     parser = argparse.ArgumentParser()
+#     parser.add_argument("--device", default="0", type=str)
+#     parser.add_argument(
+#         "--target_layer",
+#         default="layer4",
+#         type=str,
+#         help="Specify the name of the target layer (before the pooling layer)",
+#     )
+#     parser.add_argument(
+#         "--final_layer", default="fc", type=str, help="Specify the name of the last classification layer"
+#     )
+#     args = parser.parse_args()
+#     device = torch.device("cuda:" + args.device) if torch.cuda.is_available() else torch.device("cpu")
+#     model = torch.load("temp/resnet-cam.pt", map_location=device)
+#     # print(model)
+#     if torch.cuda.is_available():
+#         model.cuda()
+#     model.eval()
+#     # cam_computer = CAM(model, target_layers=[args.target_layer, args.final_layer])
+#     cam_computer = GradCAM(model, target_layers=args.target_layer)
+#     resize_param = (224, 224)
+#     norm_mean = [0.5528, 0.5528, 0.5528]
+#     norm_std = [0.1583, 0.1583, 0.1583]
+#     disp_size = 10
+#     preprocess = transforms.Compose(
+#         [transforms.Resize(resize_param), transforms.ToTensor(), transforms.Normalize(norm_mean, norm_std)]
+#     )
+#     plt.figure(figsize=(disp_size, disp_size))
+#     for i, file in enumerate(glob.glob("./temp/test_images/*")):
+#         image = PIL.Image.open(file)
+#         h, w, b = np.shape(np.array(image))
+#         img_tensor = preprocess(image).unsqueeze(0).to(device)
+#         cam_img = cam_computer(img_tensor)[0].detach().cpu().numpy()
```

Comment:
Normalizing cam_img is preferred here. Something as simple as the following should work:

`cam_img = (cam_img-cam_img.min())/ cam_img.max()`